### DISCUSSION TITLE
Which parts of the code deals with tab numbers and navigation?

### DISCUSSION THREAD

**Discussion by Aster89** [qutebrowser/qutebrowser#7238] [Q&A] [answered]
(For the purpose of understanding if I can contribute...)

Which parts of the code should I look at to see how `tab-next` and `tab-prev` are implemented, and where the indices of the tabs are generated.

**Comment by The-Compiler** [comment, marked as answer]
> (For the purpose of understanding if I can contribute...)

What are you planning to work on?

> Which parts of the code should I look at to see how `tab-next` and `tab-prev` are implemented

The `tab_next` and `tab_prev` functions in `qutebrowser/browser/commands.py`

> and where the indices of the tabs are generated.

By Qt's `QTabWidget`

**Reply by Aster89** [reply]
Thinking about the equivalent of `set rnu`... otherwise `[count]J` and `[count]K` are useless beyond ~5. And by the way, I see I have no idea how I can call `:tab-next`/`:tab-prev` with a count :disappointed:

**Reply by The-Compiler** [reply]
> Thinking about the equivalent of `set rnu`... otherwise `[count]J` and `[count]K` are useless beyond ~5.

It sounds like you want `{relative_index}` in `tabs.title.format` then? That already exists :wink: 

> And by the way, I see I have no idea how I can call `:tab-next`/`:tab-prev` with a count

Via keybindings, just as you outlined above: Type the count before the keybinding. Outside of keybindings, there is `run-with-count`.

**Reply by Aster89** [reply]
Aaaaaaaa, I feel so naive :cry:

**Comment by Aster89** [comment]
Is it expected behavior that pinned tabs show the absolute number?

**Reply by The-Compiler** [reply]
I'm guessing you changed `tabs.title.format` but not `tabs.title.format_pinned`?

**Reply by Aster89** [reply]
Obviously you're right.

**Comment by Aster89** [comment]
When there are many tabs, something like this happens
![one](https://user-images.githubusercontent.com/20521900/188419867-de27012a-1ee9-438c-a99e-ac200629bf31.png)
![another](https://user-images.githubusercontent.com/20521900/188419891-fe5a0456-7436-497f-be3c-388866f798a6.png)
i.e. the relative number is not shown even if technically there is room for it.

In fact, when there is room
![room](https://user-images.githubusercontent.com/20521900/188420130-0fdec80b-874c-4598-9bd4-398d284a2b9f.png)
the relative number seems to be aligned in a different way than the absolute number.

Is this a misconfiguration on my side?

**Reply by toofar** [reply]
Not a complete answer, but have a look at this discussion and see if it helps: https://github.com/qutebrowser/qutebrowser/issues/6927
